puppet/zulip/templates/uwsgi.ini.template.erb

```diff
@@ -7,18 +7,31 @@ env=LANG=C.UTF-8
 uid=zulip
 gid=zulip
 
-# Where the main application is located
-chdir=/home/zulip/deployments/current/
+# We run the main process in /, and chdir to the current "current"
+# directory in each forked process
+chdir=/
+hook-post-fork=chdir:/home/zulip/deployments/current
+
+# lazy-apps means that the application is loaded after forking (and
+# thus chdir'ing).  This lets us do rolling restarts into new
+# versions.
+# https://uwsgi-docs.readthedocs.io/en/latest/articles/TheArtOfGracefulReloading.html#preforking-vs-lazy-apps-vs-lazy
+lazy-apps=true
```

Comment:
Is there a downside to doing this unconditionally? Previously it was limited to when we had the setting enabled.

Reply:
The nominal reason was that [adding `lazy-apps` increases your memory footprint](https://uwsgi-docs.readthedocs.io/en/latest/articles/TheArtOfGracefulReloading.html#preforking-vs-lazy-apps-vs-lazy).  However, because [Django defers all of its work until later](https://uwsgi-docs.readthedocs.io/en/latest/articles/TheArtOfGracefulReloading.html#dealing-with-ultra-lazy-apps-like-django), we weren't actually getting any memory-sharing from loading in the main process, so shifting to everyone having `lazy-apps` is primarily for consistency.

I attempted to address this in the commit message's meanderings about memory usage, but I'll split it out into its own commit for clarity.